mmv1/api/resource.go

```diff
@@ -1279,15 +1280,30 @@ func CompareByName(a, b *Type) int {
 	return strings.Compare(a.Name, b.Name)
 }
 
-func (r Resource) GetPropertyUpdateMasksGroups() map[string][]string {
+func (r Resource) GetPropertyUpdateMasksGroupKeys(properties []*Type) []string {
+	keys := []string{}
+	for _, prop := range properties {
+		if prop.FlattenObject {
+			k := r.GetPropertyUpdateMasksGroupKeys(prop.Properties)
+			keys = append(keys, k...)
+		} else if len(prop.UpdateMaskFields) > 0 {
+			keys = append(keys, google.Underscore(prop.Name))
+		} else {
+			keys = append(keys, google.Underscore(prop.Name))
+		}
```

Comment:
These two cases can be combined to one?

Reply:
yeah, they started off different but ended up being the same later